asv_bench/benchmarks/join_merge.py

```diff
@@ -313,22 +316,52 @@ def setup(self, direction):
         self.df2d = df2[["time32", "value2"]]
         self.df1e = df1[["time", "key", "key2", "value1"]]
         self.df2e = df2[["time", "key", "key2", "value2"]]
+        self.df1f = df1[["timeu64", "value1"]]
+        self.df2f = df2[["timeu64", "value2"]]
+
+    def time_on_int(self, direction, tolerance):
```

Comment:
Instead of separate functions like this can you also just parametrze on `dtype`?

Reply:
Do you want all of the functions to implement the dtype, or only do this for the `time_on` functions?

The option to leave the dtype out for the others would be for example to negate it using the conventional `_` in the list of arguments, but I think this wouldn't be the cleanest of cases?

```suggestion
    def time_on(self, dtype, direction, tolerance):
        self.df1a["time"] = self.df1a.time.astype(dtype)
        self.df2a["time"] = self.df2a.time.astype(dtype)
        ...
        
    def time_by_object(self, _, direction, tolerance):
        ...
        )
```

Reply:
Hmm OK I see your point. I think OK to leave as is then